### PR TITLE
feat: Add initial login interface ONLY.. no login yet...

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

This PR adds an initial login interface only without any login or interaction.

### CODE REVIEW

File: src/support_sphere/lib/auth/login/login.dart [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                GestureDetector(
+                  onTap: () {
+                    Navigator.of(context).pop();
+                  },
+                  child: const IconLogo(),
+                ),
+                const SizedBox(height: 10.0),
+                Center(
+                  child: Text(
+                    'Log into your account and get started!',
```
Comment (nikiburggraf) [inline src/support_sphere/lib/auth/login/login.dart:39]: A person could be logging into their account after having gotten started. Maybe just `'Log into an existing account'`

-----

I wonder if there's a better way for these and other strings to be stored elsewhere and referenced here? Could be overkill, but if we could figure that out now, it's possible that we can make a prototype that could more easily support multiple languages in the future

File: src/support_sphere/lib/auth/login/login.dart [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            // highlightElevation: 4.0,
+            child: const Text(
+              "LOGIN",
+              style: TextStyle(
+                color: Colors.white,
+                fontSize: 12.0,
+                fontWeight: FontWeight.w600,
+              ),
+            ),
+            onPressed: () => {print("Login submitted")},
```
Comment (nikiburggraf) [inline src/support_sphere/lib/auth/login/login.dart:137]: Are these print statements just placeholders for now?

File: src/support_sphere/lib/auth/register/register.dart [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                ),
+              ),
+              const SizedBox(height: 30.0),
+              buildForm(context),
+              const SizedBox(height: 30.0),
+              Row(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: [
+                  const Text(
+                    'Already have an account  ',
```
Comment (nikiburggraf) [inline src/support_sphere/lib/auth/register/register.dart:41]: nit: `'Already have an account  '` -> `'Already have an account?'`

**Comment by nikiburggraf** [comment]
No idea how to solve this, but also seeing some weird font/text spacing things, especially on Firefox (Chrome is a bit better).

On both, the "Create a new account...." text has some wild spacing in it

Chrome is on the left, Firefox on the right in the screenshots below

Screenshot 2024-08-13 at 10 16 57 AM
Screenshot 2024-08-13 at 10 16 22 AM

**Review by nikiburggraf — COMMENTED** [review]
Couple comments and questions on this

File: src/support_sphere/lib/auth/login/login.dart (hunk 2 above)
Comment (lsetiawan) [reply src/support_sphere/lib/auth/login/login.dart:137]: Yes indeed 😄

File: src/support_sphere/lib/auth/login/login.dart [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+  const Login({Key? key}) : super(key: key);
+
+  @override
+  _LoginState createState() => _LoginState();
+}
+
+class _LoginState extends State<Login> {
+  @override
+  Widget build(BuildContext context) {
+    // LoginViewModel viewModel = Provider.of<LoginViewModel>(context);
```
Comment (nikiburggraf) [inline src/support_sphere/lib/auth/login/login.dart:19]: Add a todo for when we're gonna use this maybe? (& other comments)

or remove, up to you :)

File: src/support_sphere/lib/auth/login/login.dart [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+          ),
+        ),
+      ],
+    ));
+  }
+
+  void thisFunc() {
+    print('Forgot password clicked');
+  }
+}
```
Comment (lsetiawan) [inline src/support_sphere/lib/auth/login/login.dart:147]:
```suggestion
}
```

File: src/support_sphere/lib/auth/register/register.dart (hunk 3 above)
Comment (lsetiawan) [inline src/support_sphere/lib/auth/register/register.dart:41]:
```suggestion
                    'Already have an account? ',
```

File: src/support_sphere/lib/auth/login/login.dart [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  const Login({Key? key}) : super(key: key);
+
+  @override
+  _LoginState createState() => _LoginState();
+}
+
+class _LoginState extends State<Login> {
+  @override
+  Widget build(BuildContext context) {
+    // LoginViewModel viewModel = Provider.of<LoginViewModel>(context);
```
Comment (lsetiawan) [inline src/support_sphere/lib/auth/login/login.dart:19]:
```suggestion
    // TODO: LoginViewModel viewModel = Provider.of<LoginViewModel>(context);
    // Need to create a view model and controller for login
```

File: src/support_sphere/lib/auth/login/login.dart [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+    return LoadingOverlay(
+        isLoading: false,
+        child: Scaffold(
+          backgroundColor: Theme.of(context).colorScheme.onPrimary,
+          // key: viewModel.scaffoldKey,
```
Comment (lsetiawan) [inline src/support_sphere/lib/auth/login/login.dart:24]:
```suggestion
          // TODO: key: viewModel.scaffoldKey,
```

File: src/support_sphere/lib/auth/login/login.dart [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+                GestureDetector(
+                  onTap: () {
+                    Navigator.of(context).pop();
+                  },
+                  child: const IconLogo(),
+                ),
+                const SizedBox(height: 10.0),
+                Center(
+                  child: Text(
+                    'Log into your account and get started!',
```
Comment (lsetiawan) [inline src/support_sphere/lib/auth/login/login.dart:40]:
```suggestion
                    'Log into an existing account',
```

File: src/support_sphere/lib/auth/login/login.dart (hunk 1 above)
Comment (lsetiawan) [reply src/support_sphere/lib/auth/login/login.dart:40]: @nikiburggraf and @lsetiawan has chatted in person and will be storing strings to a `string_catalog` module in future changes.

**Comment by lsetiawan** [comment]
Thanks @nikiburggraf for your review. We'll punt the weirdness of text styling to the future.. potentially trying out the Google Fonts method.

**Review by nikiburggraf — APPROVED** [review]
:shipit:

**Pull request merged**